pkg/controller/servicebindingrequest/retriever.go

```diff
@@ -141,6 +141,67 @@ func (r *Retriever) read(place, path string, xDescriptors []string) error {
 	return nil
 }
 
+func (r *Retriever) readAnnotation(key, value string) error {
```

Comment:
I see that `readAnnotation` is turning out to be a duplication of `read` method. How could we make this generic enough so we don't need to copy-and-paste logic?

Reply:
From the usage of both, I don't think they should be unified: `read` reads OLM descriptor related data, and `readAnnotation` returns one of the CR's annotations.

Either way, _read_ is not a descriptive for those methods; I'd use _get_ instead, for example _getOLMDescriptors_ and _getAnnotation_.